### DISCUSSION TITLE
Async and jumphost

### DISCUSSION THREAD

**Discussion by joaosousa997** [ktbyers/netmiko#3088] [Q&A] [answered]
Is it viable to use async to run commands on multiple hosts based on a connection to a jump server?

Right now what i have is:

```
jump_server = {
        'host': 'x.x.x.x',
        'username': username,
        'password': password,
        'device_type': 'terminal_server',
        "fast_cli": False,
        "global_delay_factor": 2,
        'allow_auto_change': True
        }

jump_conn = ConnectHandler(**jump_server)
server = jump_conn.find_prompt()
print("-----------------INSIDE SERVER: {}-----------------".format(server))


for ip in ips:
            tasks.append(asyncio.create_task(send_commands(jump_conn ,ip)))
        await asyncio.gather(*tasks)

async def send_commands(jump_conn, route_ip):
    try:
        # create connection to router through jumphost
        jump_conn.write_channel(ssh -o StrictHostKeyChecking=no {route_ip}\n')
        ... passoword and commands...
```

Does using async will achieve my goal of running multiple instances of SSH to the Hosts and do send_command/write_channel?

Or will i need to use something like threading?

**Comment by ktbyers** [comment, marked as answer]
Yeah, I don't know if that would work (I somewhat doubt it). You would have to test it out and see if you achieved the behavior you wanted. The standard Netmiko solution is to use threads or multiple processes.

**Reply by joaosousa997** [reply]
hey! Sorry for the delay, i ended up using multi processing for this and it works great :)
